mods/beds/api.lua

```diff
@@ -12,15 +10,14 @@ local function destruct_bed(pos, n)
 		local dir = minetest.facedir_to_dir(node.param2)
 		other = vector.add(pos, dir)
 	end
-
-	if reverse then
-		reverse = not reverse
-		minetest.remove_node(other)
-		minetest.check_for_falling(other)
-		beds.remove_spawns_at(pos)
-		beds.remove_spawns_at(other)
-	else
-		reverse = not reverse
+	local odef = minetest.registered_nodes[minetest.get_node(other).name]
+	if odef.groups.bed > 0 then
```

Comment:
This is error-prone: If `minetest.registered_nodes[minetest.get_node(other).name]` is `nil` (which is the case for undefined nodes) this will crash. I recommend using `minetest.get_item_group(minetest.get_node(other).name, "bed") ~= 0` instead.